## EDA - The Price of Art

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import kagglehub
import os

In [18]:
# Download latest version
path = kagglehub.dataset_download("flkuhm/art-price-dataset")

print("Path to dataset files:", path)

Path to dataset files: /home/codespace/.cache/kagglehub/datasets/flkuhm/art-price-dataset/versions/1


In [19]:
# 1. First, let's see what files are inside the directory we just downloaded.
print("Files in the directory:", os.listdir(path))

Files in the directory: ['artDataset.csv', 'artDataset']


In [20]:
# 2. Now, we need the exact filename. Let's assume from the Kaggle page that the file is named 'artDataset'
filename = 'artDataset.csv'

In [21]:
# 3. We create the full, correct path to the file. 
# os.path.join is the best way to do this, as it works on any operating system (Windows, Mac, Linux).
full_path = os.path.join(path, filename)

In [22]:
# 4. Finally, we read the CSV file into our pandas DataFrame.
try:
    df = pd.read_csv(full_path)
    print("\nDataset loaded successfully!")
except FileNotFoundError:
    print(f"\nError: The file '{filename}' was not found in the directory '{path}'.")
    print("Please check the filename from the 'os.listdir(path)' output and update the 'filename' variable.")



Dataset loaded successfully!


In [23]:
df = pd.read_csv(full_path)
df

,Unnamed: 0,price,artist,title,yearCreation,signed,condition,period,movement
0,0,28.500 USD,Tommaso Ottieri,Bayreuth Opera,2021,Signed on verso,This work is in excellent condition.,Contemporary,Baroque
1,1,3.000 USD,Pavel Tchelitchew,Drawings of the Opera,First Half 20th Century,Signed and titled,Not examined out of frame.No obvious signs of ...,Post-War,Surrealism
2,2,5.000 USD,Leo Gabin,Two on Sidewalk,2016,"Signed, titled and dated on verso",This work is in excellent condition.,Contemporary,Abstract
3,3,5.000 USD,Matthias Dornfeld,Blumenszene,2010,"Signed, titled and dated on the reverse with t...",This work is in excellent condition.There is m...,Contemporary,Abstract
4,4,2.500 USD,Alexis Marguerite Teplin,Feverish Embarkation,2001,Signed on verso,This work is in excellent condition.,Contemporary,Abstract
...,...,...,...,...,...,...,...,...,...
749,749,680 USD,Jane Kent,Miracle Grow #17,2012,Signed and dated on lower right.,Not examined out of frame.No obvious signs of ...,Contemporary,Abstract
750,750,1.275 USD,Gary Bower,Rolph Series,1970,[nan],Not examined out of frame.Significant undulati...,Contemporary,Geometric Abstraction
751,751,680 USD,Jane Kent,Untitled,2012,[nan],Not examined out of frame.No apparent imperfec...,Contemporary,Geometric Abstraction
752,752,1.275 USD,T. L. Solien,Juniper,1986,[nan],Not examined outside of frame.Pinholes at edge...,Contemporary,Abstract


In [24]:
# Let's understand the dimensions and the columns of our dataset.
# This gives us a quick overview without printing thousands of rows.

print(f"The DataFrame has {df.shape[0]} rows and {df.shape[1]} columns.")
print("\nColumn Names:")
print(df.columns.tolist())

The DataFrame has 754 rows and 9 columns.

Column Names:
['Unnamed: 0', 'price', 'artist', 'title', 'yearCreation', 'signed', 'condition', 'period', 'movement']


In [25]:
# The 'Unnamed: 0' column is likely an old index from the CSV file.
# It's redundant, so let's drop it to clean our DataFrame.

if 'Unnamed: 0' in df.columns:
    df.drop('Unnamed: 0', axis=1, inplace=True)
    print("Dropped the 'Unnamed: 0' column.")
else:
    print("'Unnamed: 0' column not found. No action taken.")

# Let's verify the change
print("\nUpdated Column Names:")
print(df.columns.tolist())

Dropped the 'Unnamed: 0' column.

Updated Column Names:
['price', 'artist', 'title', 'yearCreation', 'signed', 'condition', 'period', 'movement']


In [26]:
# Get a concise summary of the DataFrame to check data types and missing values.
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 754 entries, 0 to 753
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   price         754 non-null    str  
 1   artist        753 non-null    str  
 2   title         754 non-null    str  
 3   yearCreation  754 non-null    str  
 4   signed        754 non-null    str  
 5   condition     754 non-null    str  
 6   period        754 non-null    str  
 7   movement      754 non-null    str  
dtypes: str(8)
memory usage: 47.3 KB


In [27]:
# Get descriptive statistics for the numerical columns.
df.describe().T

,count,unique,top,freq
price,754,108,800 USD,124
artist,753,454,Russell Young,17
title,754,679,Untitled,35
yearCreation,754,136,2012,34
signed,754,390,[nan],153
condition,754,376,Excellent condition.,82
period,754,5,Contemporary,414
movement,754,34,Realism,177


## Initial Findings from Data Inspection
After the initial data loading and inspection using .info() and .describe(), several key characteristics and potential issues have been identified in the dataset:

- Data Type Mismatch: All columns, including critical ones like price and yearCreation, are currently stored as objects (str). This prevents any numerical or time-series analysis.
- Price Column: The price column contains the ' USD' suffix, which needs to be removed before conversion to a numeric type. The most frequent price is '$800', which may indicate a specific category of art or a default price that requires further investigation.
Missing Values:
- The artist column has one official missing value (NaN). Given that there is only one, this warrants a specific investigation of the corresponding title to make an informed imputation decision (e.g., is it a famous work, or by an 'Anonymous' artist?).
- The signed column contains the literal string '[nan]' as its most frequent value. This is a hidden missing value that needs to be converted to a proper NaN for accurate analysis.
- High Cardinality: The condition column has 376 unique values, which is too many for effective analysis. It will likely require grouping or simplification later in the project.

## Proposed Cleaning & Preprocessing Plan
To prepare the dataset for exploratory analysis and visualization, the following steps will be taken in order:

- Investigate Missing Artist: Before any bulk transformations, we will isolate the single row with the missing artist value. By examining its title and other attributes, we will decide on the most appropriate value to impute (e.g., "Anonymous", "Unknown", or the actual artist's name if identifiable).
- Convert price to Numeric:
Remove the ' USD' suffix and any commas from the price column.
Convert the column to a numeric (float) data type.
- Convert yearCreation to Numeric:
Inspect the column for non-numeric characters.
Convert the column to a numeric (integer) data type.
- Handle Hidden Missing Values:
Convert the literal '[nan]' strings in the signed column to actual NaN values.
Decide on a strategy for the now-missing values in the signed column (e.g., imputation with a placeholder like 'Not Signed').
- (Future) Simplify Categorical Columns:
Once the main structural issues are resolved, we will address the high cardinality in the condition column to make it more analyzable.

## STEP I - Investigating the single missing 'artist' value:

In [28]:
# Let's find the row where the 'artist' is missing (NaN)
missing_artist_row = df[df['artist'].isnull()]

# Display the full information for this row
print("Investigating the row with the missing artist:")
print(missing_artist_row.T) # Using .T to transpose it for better readability

Investigating the row with the missing artist:
                                                            725
price                                                 1.275 USD
artist                                                      NaN
title                                                     [nan]
yearCreation                                              [nan]
signed                      Signed and dated in pencil to verso
condition     Not examined out of frame.Minor sheet undulati...
period                                             Contemporary
movement                                                Realism


In [29]:
# Based on our investigation of row 725:
# We will impute missing values for 'artist' and 'title' to avoid data loss, as the row contains valid price and category information.

df.loc[df['artist'].isnull(), 'artist'] = 'Unknown Artist'
df.loc[df['title'] == '[nan]', 'title'] = 'Untitled / Unknown'

# Verify the fix
print(df.loc[725, ['artist', 'title', 'price']])

artist        Unknown Artist
title     Untitled / Unknown
price              1.275 USD
Name: 725, dtype: str


## Handling Specific Missing Values
During the initial inspection, a specific case (index 725) was identified where the 'artist' and 'title' were missing, yet the record contained valid transaction data (price) and stylistic metadata (Contemporary Realism).

Instead of removing this entry—which would bias our price analysis—I decided to impute these missing categorical fields with 'Unknown Artist' and 'Untitled / Unknown'. This preserves the financial data for the Power BI dashboard while maintaining the integrity of the categorical filters.

## STEP II - Price Sanitization

In [30]:
# We create a copy to avoid SettingWithCopyWarning and keep the original for reference if needed
df['price'] = df['price'].str.replace(' USD', '', regex=False) # Remove suffix
df['price'] = df['price'].str.replace('.', '', regex=False)    # Remove thousand separator (.)
df['price'] = df['price'].astype(float)                       # Convert to numeric

# Let's check the statistics to make sure the prices make sense now
print("Price Statistics:")
print(df['price'].describe())

Price Statistics:
count       754.000000
mean       6760.769231
std       12828.990543
min         595.000000
25%        1000.000000
50%        2500.000000
75%        7000.000000
max      160000.000000
Name: price, dtype: float64


In [32]:
print(df['price'])

0      28500.0
1       3000.0
2       5000.0
3       5000.0
4       2500.0
        ...   
749      680.0
750     1275.0
751      680.0
752     1275.0
753      680.0
Name: price, Length: 754, dtype: float64


## Data Normalization: Price Column:
The price column was originally loaded as a string object, containing currency suffixes ('USD') and period-based thousand separators (e.g., '1.275').

To enable quantitative analysis and ensure compatibility with Power BI's numerical schemas, I performed a sanitization process:

1.- Stripped the 'USD' currency descriptor.

2.- Removed the period separators to prevent Python from interpreting them as decimal points (as the dataset follows a format where 1.275 equals 1,275).

3.- Cast the resulting strings into a float data type.

This transformation allows for the calculation of total market value, average pricing per movement, and other key financial metrics."

In [33]:
# Convert to integer since we don't have decimals
df['price'] = df['price'].astype(int)

# Now let's check the head to see them nice and clean
print(df['price'].head())

0    28500
1     3000
2     5000
3     5000
4     2500
Name: price, dtype: int64


Refining Price Data Type
After the initial cleaning of the price column, a statistical review revealed that all transaction values in the dataset were whole numbers (integers), with no decimal precision required.

To improve data readability and optimize memory usage, I converted the price column from float to int. This decision ensures that the EDA reflects the actual nature of the art market entries in this dataset and simplifies the visual representation in the final Power BI dashboard.

## STEP III - GLOBAL NULL STANDARDIZATION

In [34]:
# We replace the literal string '[nan]' with the official NumPy representation of Null
df.replace('[nan]', np.nan, inplace=True)

,price,artist,title,yearCreation,signed,condition,period,movement
0,28500,Tommaso Ottieri,Bayreuth Opera,2021,Signed on verso,This work is in excellent condition.,Contemporary,Baroque
1,3000,Pavel Tchelitchew,Drawings of the Opera,First Half 20th Century,Signed and titled,Not examined out of frame.No obvious signs of ...,Post-War,Surrealism
2,5000,Leo Gabin,Two on Sidewalk,2016,"Signed, titled and dated on verso",This work is in excellent condition.,Contemporary,Abstract
3,5000,Matthias Dornfeld,Blumenszene,2010,"Signed, titled and dated on the reverse with t...",This work is in excellent condition.There is m...,Contemporary,Abstract
4,2500,Alexis Marguerite Teplin,Feverish Embarkation,2001,Signed on verso,This work is in excellent condition.,Contemporary,Abstract
...,...,...,...,...,...,...,...,...
749,680,Jane Kent,Miracle Grow #17,2012,Signed and dated on lower right.,Not examined out of frame.No obvious signs of ...,Contemporary,Abstract
750,1275,Gary Bower,Rolph Series,1970,NaN,Not examined out of frame.Significant undulati...,Contemporary,Geometric Abstraction
751,680,Jane Kent,Untitled,2012,NaN,Not examined out of frame.No apparent imperfec...,Contemporary,Geometric Abstraction
752,1275,T. L. Solien,Juniper,1986,NaN,Not examined outside of frame.Pinholes at edge...,Contemporary,Abstract


In [35]:
# Let's verify how many ACTUAL missing values we have now in each column
print("Updated Missing Values Count:")
print(df.isnull().sum())

Updated Missing Values Count:
price             0
artist            0
title             0
yearCreation     19
signed          153
condition         7
period            1
movement          4
dtype: int64


In [36]:
# Let's see specifically the columns we were worried about
print("\nCheck for specific columns:")
print(df[['yearCreation', 'signed', 'condition']].head(10))


Check for specific columns:
               yearCreation  \
0                      2021   
1  First Half 20th Century    
2                      2016   
3                      2010   
4                      2001   
5                      1977   
6                      2007   
7                      1993   
8                      2014   
9                      1993   

                                              signed  \
0                                    Signed on verso   
1                                  Signed and titled   
2                  Signed, titled and dated on verso   
3  Signed, titled and dated on the reverse with t...   
4                                    Signed on verso   
5  Signed lower right recto; numbered lower left ...   
6        Numbered and signed on bottom corner, recto   
7  Signed lower right recto; numbered lower left ...   
8  Signed lower right recto; numbered lower left ...   
9  Signed lower right recto; numbered lower left ...   

            

# Data Quality Audit & Initial Findings
Upon standardizing the dataset and converting pseudo-null strings ([nan]) into actual null objects, a clear picture of the data quality emerged:

1.- Year of Creation (yearCreation): Only 19 records are completely missing. However, the column contains a mix of specific years (e.g., '2021') and categorical periods (e.g., 'First Half 20th Century'), requiring a custom parsing strategy.

2.- Authentication (signed): This is the most incomplete field with 153 missing values. This suggests that while price and artist are well-documented, the physical details of the signature are often omitted in the catalog.

3.- Physical State (condition): With only 7 missing values, this column is remarkably complete, though it exhibits high textual complexity that will require grouping.

4.- Artistic Context (period & movement): These fields are nearly perfect, with only 1 and 4 missing values respectively, providing a solid foundation for categorical analysis.

# Strategic Decision: Data Recovery over Deletion
Instead of dropping rows with missing or descriptive temporal data, I have opted for a Data Recovery approach:

1.- Imputation by Context: For the 19 missing years, I will leverage the movement and period columns to estimate the era.

2.- Mapping Categorical Periods: Descriptions like 'First Half 20th Century' will be mapped to their chronological mid-points (e.g., 1925) to maintain a continuous timeline for Power BI visualizations.

3.- Labeling: Missing values in the signed column will be labeled as 'Not Specified' to serve as a valid filter category rather than a 'Blank' entry.